src/jslogger.js

```diff
@@ -0,0 +1,322 @@
+            xhr.success(function() {
+                xhrStatus('post request success', req_url, xhr, startTime, true);
+            });
+
+            xhr.fail(function() {
+                xhrStatus('post request failed', req_url, xhr, startTime, false);
+            });
+
+            return xhr;
+        }
+
+        // flush logs into server
+        function ajaxCall() {
+
+            if(isAjaxCompleted) {
+                flushPendingLogs();
+            }
+        }
+
+        /**
+         * Send pending logs to url which is provided by checking logs that are being logged.
+         */
+        function flushPendingLogs() {
+
+            isAjaxCompleted  = false;
+            var pending_logs = 0,
+                data         = [],
+                log_keys     = [];
+
+            for(var i in window.localStorage){
+                pending_logs = i.startsWith('logging') ? pending_logs + 1 : pending_logs;
+            }
+
+            if(pending_logs == 0) {
+                isAjaxCompleted = true;
+                return;
+            }
+
+            for(var key_index in window.localStorage) {
+                if(key_index.startsWith('logging') && data.length < 1000) {
+                    data.push(JSON.parse(window.localStorage.getItem(key_index)));
+                    log_keys.push(key_index);
+                }
+            }
+
+            var params = {
+                log : data,
+            }
+
+            var xhr = ajax({
+                url     : url,
+                type    : 'POST',
+                data    : JSON.stringify(params),
+                timeout : 1000 * 60 * 10,
+                success : function() {
+                    for(var key_index in log_keys) {
+                        delete window.localStorage[log_keys[key_index]]
+                    }
+                },
+            });
+
+            xhr.done(function(response) {
+
+                pending_logs -= 100;
+                if(pending_logs > 0) {
+                    flushPendingLogs();
+                }
+
+                isAjaxCompleted = true;
+            });
+        }
+
+        // Time interval for sending logs to url
+        setInterval(function() {
+            ajaxCall();
+        }, timeInterval);
+
+        // capture javascript errors
+        window.onerror = function(message, source, lineno, colno, error) {
+
+            var time = new Date(),
+                data = {
+                "UUID"      : uuidv1(),
+                "timestamp" : time.getTime(),
+                "message"   : message,
+                "url"       : source,
+                "lineno"    : lineno,
+                "colno"     : colno,
+                "error"     : error,
+                "level"     : "javascript",
+                "host_url"  : window.location.host
+            }
+
+            appender(data);
+            return;
+        }
+
+        ajaxCall();
+    }
+
+window.jslogger = jslogger;
+})();
```

Comment:
@jaswanth098 @bajibabu9 Aren't we providing a min.js to?

Reply:
@jaswanth098 For now it is okay to pregenerate and checkin. Later we must consider adding build instructions to have the user themselves generate the min file (perhaps some tool such as gulp would help - @prrashi can suggest better).

Reply:
@jaswanth098 Please consult with @prrashi about the best way to handle uuid.js (ideally this shouldn't a part of our codebase).